sacro-app/src/create-window.js

```diff
@@ -9,9 +9,30 @@ const { mainMenu } = require("./main-menu");
 const startServer = require("./start-server");
 const { waitThenLoad } = require("./utils");
 
+function windowBuilder(options) {
+  const win = new BrowserWindow({
+    width: 1024,
+    height: 768,
+    ...options,
+  });
+
+  if (options.file) {
+    win.loadFile(options.file);
+  }
+
+  return win;
+}
+
 let TEMPDIR = null;
 
 const createWindow = async () => {
+  const error = windowBuilder({ file: "error.html", show: false });
+  const splash = windowBuilder({
```

Comment:
Huh, this looks very different to what I'd expected. We have 3 separate windows?

I'd naively assumed we would:

1) create the window
2) `win.loadUrl("file:///path/to/splash.html")`
3) wait for server
 - on success `win.loadUrl(serverUrl)`
 - on error `win.loadUrl("file:///path/to/error.html")`

Does that not work?

Reply:
Hah!  Hadn't even occurred to me to test changing the loaded resource for the current window.  That works and simplifies some things 👍

Reply:
It's the much vaunted programmable hypermedia client! :magic_wand: